api/v1/server/middleware/webhook_rate_limit.go

```diff
@@ -0,0 +1,43 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/rs/zerolog"
+	"golang.org/x/time/rate"
+)
+
+func WebhookRateLimitMiddleware(rateLimit rate.Limit, burst int, l *zerolog.Logger) echo.MiddlewareFunc {
+	config := middleware.RateLimiterConfig{
+		Skipper: func(c echo.Context) bool {
+			method := c.Request().Method
+
+			if method != http.MethodPost {
+				return true
+			}
+
+			tenantId := c.Param("tenant")
+			webhookName := c.Param("v1-webhook")
+
+			if tenantId == "" || webhookName == "" {
+				return true
+			}
+
+			return c.Request().URL.Path != fmt.Sprintf("/api/v1/stable/tenants/%s/webhooks/%s", tenantId, webhookName)
```

Comment:
This seems pretty brittle, but will add a comment in the OpenAPI spec pointing at this

Reply:
Is there no way to use the hatchetmiddleware which has a signature of `Middleware(r *middleware.RouteInfo) echo.HandlerFunc`? The `RouteInfo` will give us a more stable operation ID (which is still brittle, but perhaps not as brittle)

Reply:
maybe we can? I'll look into that quickly 🙌